### ISSUE TITLE
Issue on c2me and lithostitched after Update (Fabric 1.21.1 / 0.16.9)

### ISSUE THREAD

**Issue by suerion** [Apollounknowndev/lithostitched#19]
Hey, i got on Worldgenerating some errors, had this issue not on latest version

latest.log 
https://mclo.gs/a6cehBp
```
[13:31:32 ERROR] [c2me-worker-0]: [Chunk System of witchery:nightmare_world] Error upgrading chunk [-121, -124] to "minecraft:structure_starts"
java.util.ConcurrentModificationException: null
        at java.base/java.util.ArrayList.sort(ArrayList.java:1806) ~[?:?]
        at knot/dev.worldgen.lithostitched.worldgen.structure.LithostitchedTemplates.shuffle(LithostitchedTemplates.java:28) ~[lithostitched-fabric-1.21.1-1.3.7.jar:?]
        at knot/dev.worldgen.lithostitched.worldgen.structure.AlternateJigsawGenerator$StructurePoolGenerator.getPoolElements(AlternateJigsawGenerator.java:222) ~[lithostitched-fabric-1.21.1-1.3.7.jar:?]
        at knot/dev.worldgen.lithostitched.worldgen.structure.AlternateJigsawGenerator$StructurePoolGenerator.findAndTestChildCandidates(AlternateJigsawGenerator.java:186) ~[lithostitched-fabric-1.21.1-1.3.7.jar:?]
        at knot/dev.worldgen.lithostitched.worldgen.structure.AlternateJigsawGenerator$StructurePoolGenerator.generatePiece(AlternateJigsawGenerator.java:177) ~[lithostitched-fabric-1.21.1-1.3.7.jar:?]
        at knot/dev.worldgen.lithostitched.worldgen.structure.AlternateJigsawGenerator.generate(AlternateJigsawGenerator.java:128) ~[lithostitched-fabric-1.21.1-1.3.7.jar:?]
        at knot/dev.worldgen.lithostitched.worldgen.structure.AlternateJigsawGenerator.lambda$generate$3(AlternateJigsawGenerator.java:99) ~[lithostitched-fabric-1.21.1-1.3.7.jar:?]
        at knot/net.minecraft.world.level.levelgen.structure.Structure$GenerationStub.lambda$getPiecesBuilder$0(Structure.java:152) ~[?:?]
        at knot/com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[datafixerupper-8.0.16.jar:?]
        at knot/net.minecraft.world.level.levelgen.structure.Structure$GenerationStub.getPiecesBuilder(Structure.java:150) ~[?:?]
        at knot/net.minecraft.world.level.levelgen.structure.Structure.generate(Structure.java:162) ~[?:?]
        at knot/net.minecraft.world.level.chunk.ChunkGenerator.tryGenerateStructure(ChunkGenerator.java:517) ~[?:?]
        at knot/net.minecraft.world.level.chunk.ChunkGenerator.lambda$createStructures$14(ChunkGenerator.java:499) ~[?:?]
        at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
        at knot/net.minecraft.world.level.chunk.ChunkGenerator.createStructures(ChunkGenerator.java:452) ~[?:?]
        at knot/net.minecraft.world.level.chunk.status.ChunkStatusTasks.generateStructureStarts(ChunkStatusTasks.java:38) ~[?:?]
        at knot/net.minecraft.world.level.chunk.status.ChunkStep.mixinextras$bridge$doWork$14(ChunkStep.java) ~[?:?]
        at knot/net.minecraft.world.level.chunk.status.ChunkStep.wrapOperation$bmc000$c2me-opts-chunk-access$runGenerationTask(ChunkStep.java:526) ~[?:?]
        at knot/net.minecraft.world.level.chunk.status.ChunkStep.apply(ChunkStep.java:33) ~[?:?]
        at knot/com.ishland.c2me.rewrites.chunksystem.common.statuses.VanillaWorldGenerationDelegate.lambda$upgradeToThis$8(VanillaWorldGenerationDelegate.java:123) ~[c2me-rewrites-chunk-system-0.3.0.298+1.21.1-2365b2480070b912.jar:?]
        at knot/com.ishland.c2me.base.common.scheduler.ScheduledTask.run(ScheduledTask.java:22) [c2me-base-0.3.0+alpha.0.298+1.21.1-86625858f79bce5b.jar:?]
        at knot/com.ishland.flowsched.executor.WorkerThread.pollTasks(WorkerThread.java:57) [c2me-base-0.3.0+alpha.0.298+1.21.1-86625858f79bce5b.jar:?]
        at knot/com.ishland.flowsched.executor.WorkerThread.run(WorkerThread.java:27) [c2me-base-0.3.0+alpha.0.298+1.21.1-86625858f79bce5b.jar:?]
[13:31:32 WARN] [Server thread]: [ReadFromDiskAsync] Not saving partially generated broken chunk [-121, -124]
[13:31:33 ERROR] [c2me-worker-5]: [Chunk System of witchery:nightmare_world] Error upgrading chunk [-121, -124] to "minecraft:structure_starts"
java.util.ConcurrentModificationException: null
        at java.base/java.util.ArrayList.sort(ArrayList.java:1806) ~[?:?]
        at knot/dev.worldgen.lithostitched.worldgen.structure.LithostitchedTemplates.shuffle(LithostitchedTemplates.java:28) ~[lithostitched-fabric-1.21.1-1.3.7.jar:?]
        at knot/dev.worldgen.lithostitched.worldgen.structure.AlternateJigsawGenerator$StructurePoolGenerator.getPoolElements(AlternateJigsawGenerator.java:222) ~[lithostitched-fabric-1.21.1-1.3.7.jar:?]
        at knot/dev.worldgen.lithostitched.worldgen.structure.AlternateJigsawGenerator$StructurePoolGenerator.findAndTestChildCandidates(AlternateJigsawGenerator.java:186) ~[lithostitched-fabric-1.21.1-1.3.7.jar:?]
        at knot/dev.worldgen.lithostitched.worldgen.structure.AlternateJigsawGenerator$StructurePoolGenerator.findAndTestChildCandidates(AlternateJigsawGenerator.java:191) ~[lithostitched-fabric-1.21.1-1.3.7.jar:?]
        at knot/dev.worldgen.lithostitched.worldgen.structure.AlternateJigsawGenerator$StructurePoolGenerator.generatePiece(AlternateJigsawGenerator.java:177) ~[lithostitched-fabric-1.21.1-1.3.7.jar:?]
        at knot/dev.worldgen.lithostitched.worldgen.structure.AlternateJigsawGenerator.generate(AlternateJigsawGenerator.java:128) ~[lithostitched-fabric-1.21.1-1.3.7.jar:?]
        at knot/dev.worldgen.lithostitched.worldgen.structure.AlternateJigsawGenerator.lambda$generate$3(AlternateJigsawGenerator.java:99) ~[lithostitched-fabric-1.21.1-1.3.7.jar:?]
        at knot/net.minecraft.world.level.levelgen.structure.Structure$GenerationStub.lambda$getPiecesBuilder$0(Structure.java:152) ~[?:?]
        at knot/com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[datafixerupper-8.0.16.jar:?]
        at knot/net.minecraft.world.level.levelgen.structure.Structure$GenerationStub.getPiecesBuilder(Structure.java:150) ~[?:?]
        at knot/net.minecraft.world.level.levelgen.structure.Structure.generate(Structure.java:162) ~[?:?]
        at knot/net.minecraft.world.level.chunk.ChunkGenerator.tryGenerateStructure(ChunkGenerator.java:517) ~[?:?]
        at knot/net.minecraft.world.level.chunk.ChunkGenerator.lambda$createStructures$14(ChunkGenerator.java:499) ~[?:?]
        at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
        at knot/net.minecraft.world.level.chunk.ChunkGenerator.createStructures(ChunkGenerator.java:452) ~[?:?]
        at knot/net.minecraft.world.level.chunk.status.ChunkStatusTasks.generateStructureStarts(ChunkStatusTasks.java:38) ~[?:?]
        at knot/net.minecraft.world.level.chunk.status.ChunkStep.mixinextras$bridge$doWork$14(ChunkStep.java) ~[?:?]
        at knot/net.minecraft.world.level.chunk.status.ChunkStep.wrapOperation$bmc000$c2me-opts-chunk-access$runGenerationTask(ChunkStep.java:526) ~[?:?]
        at knot/net.minecraft.world.level.chunk.status.ChunkStep.apply(ChunkStep.java:33) ~[?:?]
        at knot/com.ishland.c2me.rewrites.chunksystem.common.statuses.VanillaWorldGenerationDelegate.lambda$upgradeToThis$8(VanillaWorldGenerationDelegate.java:123) ~[c2me-rewrites-chunk-system-0.3.0.298+1.21.1-2365b2480070b912.jar:?]
        at knot/com.ishland.c2me.base.common.scheduler.ScheduledTask.run(ScheduledTask.java:22) [c2me-base-0.3.0+alpha.0.298+1.21.1-86625858f79bce5b.jar:?]
        at knot/com.ishland.flowsched.executor.WorkerThread.pollTasks(WorkerThread.java:57) [c2me-base-0.3.0+alpha.0.298+1.21.1-86625858f79bce5b.jar:?]
        at knot/com.ishland.flowsched.executor.WorkerThread.run(WorkerThread.java:27) [c2me-base-0.3.0+alpha.0.298+1.21.1-86625858f79bce5b.jar:?]
```

**Comment by Apollounknowndev** [comment]
I believe I've fixed it, can you see if [this](https://drive.google.com/file/d/121tYL3Zkrb9QzuwuvtNvQM17iNBfgTD2/view?usp=sharing) version fixes the issue?

**Comment by suerion** [comment]
On worldgenerating, it looks okey now, will test a little bit more but i had found some little issues like this from the template pool

`[16:53:14 WARN] [c2me-worker-4]: [lithostitched] Template pool fallback chain found: kaisyn:village/modded/waystones/waystone_default -> kaisyn:village/modded/waystones/waystone_default`
`[16:53:36 WARN] [c2me-worker-4]: [lithostitched] Template pool fallback chain found: kaisyn:village/modded/waystones/waystone_default -> kaisyn:village/modded/waystones/waystone_default`

**Comment by Apollounknowndev** [comment]
Glad the fix works, the update will properly be out later. 
That error is meant to be a warning for mod developers (it doesn't mean anything is actually broken), alongside the next update I'll add a config option to be able to toggle it.

**Comment by suerion** [comment]
Thanks a lot :)

**Comment by Apollounknowndev** [comment]
Fixed in 1.3.8.
